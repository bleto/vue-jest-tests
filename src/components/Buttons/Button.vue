<template>
    <button
        :disabled="disabled"
        :class="buttonTypeClasses"
        :aria-label="title">
        <slot name="prepend" />
        <span
            v-if="title"
            :class="titleTypo"
            v-text="title" />
    </button>
</template>

<script>
export default {
    name: 'Button',
    props: {
        rounded: {
            type: Boolean,
            required: false,
            default: false,
        },
        fab: {
            type: Boolean,
            required: false,
            default: false,
        },
        title: {
            type: String,
            default: null,
        },
        color: {
            type: String,
            default: 'primary',
        },
        theme: {
            type: String,
            default: 'light',
        },
        disabled: {
            type: Boolean,
            default: false,
        },
        small: {
            type: Boolean,
            required: false,
            default: false,
        },
        large: {
            type: Boolean,
            required: false,
            default: false,
        },
    },
    computed: {
        titleTypo() {
            return this.small
                ? 'typo-btn--xs'
                : 'typo-btn l-spacing--half';
        },
        buttonTypeClasses() {
            return [
                'btn',
                'theme',
                `theme--${this.theme}`,
                this.color,
                {
                    'btn--small': this.small,
                    'btn--large': this.large,
                    'btn--rounded': this.rounded,
                    'btn--fab': this.fab,
                },
            ];
        },
    },
};
</script>

<style lang="scss" scoped>
    @import "@/assets/scss/button.scss";
</style>
