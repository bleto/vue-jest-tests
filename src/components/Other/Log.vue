<template>
    <div>
        <slot :log-click="onClick" :log-info="logInfo" />
    </div>
</template>

<script>
export default {
    name: 'Log',
    data() {
        return {
            logInfo: 0,
        };
    },
    methods: {
        onClick() {
            this.logInfo += 1;
            console.log('Log method');
        },
    },
};
</script>
