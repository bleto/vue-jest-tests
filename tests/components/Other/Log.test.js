import { mount, createLocalVue } from '@vue/test-utils';
import Log from '@/components/Other/Log.vue';
import Button from '@/components/Buttons/Button';

const localVue = createLocalVue();

describe('Other/Log', () => {
    let wrapper;
    beforeEach(() => {
        const buttonWrapper = {
            render(h) {
                return h(Button, { props: { title: 'Test' } });
            },
        };
        wrapper = mount(Log, {
            localVue,
            slots: {
                default: buttonWrapper,
            },
        });
    });

    it('Component is rendered', () => {
        expect(wrapper.is(Log)).toBe(true);
        expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('Component is named well', () => {
        expect(typeof Log.name).toBe('string');
        expect(Log.name).toEqual('Log');
    });

    it('Button are inserted in a Log component', () => {
        const log = wrapper.find(Log);
        expect(log.find(Button).isVueInstance()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('Checks store action click event', () => {
        const btn = wrapper.find('.btn');
        expect(btn.exists()).toBe(true);
        btn.trigger('click');
        expect(wrapper).toMatchSnapshot();
        // expect(+number.text()).toEqual(expect.any(Number));
    });
});
