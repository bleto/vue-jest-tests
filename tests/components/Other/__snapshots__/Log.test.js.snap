// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other/Log Button are inserted in a Log component 1`] = `<div><button aria-label="Test" class="btn theme theme--light primary"> <span class="typo-btn l-spacing--half">Test</span></button></div>`;

exports[`Other/Log Checks store action click event 1`] = `<div><button aria-label="Test" class="btn theme theme--light primary"> <span class="typo-btn l-spacing--half">Test</span></button></div>`;
